src/components/simulation/SimulationConfiguration.svelte

```diff
@@ -15,12 +15,23 @@
   } from '../../stores/simulation';
   import { getTarget } from '../../utilities/generic';
   import { getArguments, getFormParameters } from '../../utilities/parameters';
+  import req from '../../utilities/requests';
 
-  $: formParameters = getFormParameters(
-    $modelParametersMap,
-    $simulation?.arguments,
-    $simulation?.template?.arguments,
-  );
+  export let modelId: number;
+
+  let formParameters: FormParameter[] = [];
+
+  $: {
+    req
+      .getEffectiveModelArguments(modelId, $simulation?.arguments)
+      .then(({ arguments: defaultArgumentsMap }) => {
+        formParameters = getFormParameters(
+          $modelParametersMap,
+          $simulation?.arguments,
+          { ...defaultArgumentsMap, ...$simulation?.template?.arguments },
```

Comment:
May be nice to have a one-line/inline comment here denoting the "tiers" of arguments at play. No obligation to, I just know I get pretty confused and looking at this doesn't make it immediately clear (likely due to my TS unfamiliarity though).

Reply:
Added comment outlining the argument merging going on here